src/relocation.rs

```diff
@@ -143,6 +147,475 @@ impl ParseAt for Rela {
+            if self.group_flags.is_relocation_grouped_by_info() {
+                self.relocation.r_info = self.read_signed_leb128()? as u64;
+            }
+
+            if self.group_flags.is_relocation_grouped_by_addend()
+                && self.group_flags.is_relocation_group_has_addend()
+            {
+                if !RELA {
+                    return Err(ParseError::UnexpectedRelocationAddend);
+                }
+                self.relocation.r_addend = self
+                    .relocation
+                    .r_addend
+                    .wrapping_add(self.read_signed_leb128()?);
+            } else if !self.group_flags.is_relocation_group_has_addend() && RELA {
+                self.relocation.r_addend = 0;
+            }
+
+            self.group_relocation_index = 0;
+            Ok(())
+        }
+
+        fn read_group_relocation(&mut self) -> Result<(), ParseError> {
+            if self.group_flags.is_relocation_grouped_by_offset_delta() {
+                self.relocation.r_offset = self
+                    .relocation
+                    .r_offset
+                    .wrapping_add(self.group_r_offset_delta);
+            } else {
+                // may overflow
+                self.relocation.r_offset = self
+                    .relocation
+                    .r_offset
+                    .wrapping_add(self.read_signed_leb128()? as u64);
+            };
+
+            if !self.group_flags.is_relocation_grouped_by_info() {
+                self.relocation.r_info = self.read_signed_leb128()? as u64;
+            }
+
+            if self.group_flags.is_relocation_group_has_addend()
+                && !self.group_flags.is_relocation_grouped_by_addend()
+            {
+                if !RELA {
+                    return Err(ParseError::UnexpectedRelocationAddend);
+                }
+                self.relocation.r_addend = self
+                    .relocation
+                    .r_addend
+                    .wrapping_add(self.read_signed_leb128()?);
+            }
+
+            self.relocation_index += 1;
+            self.group_relocation_index += 1;
+
+            Ok(())
+        }
+    }
+
+    impl<'data> Iterator for ParsingIterator<'data, true> {
+        type Item = Result<Rela, ParseError>;
+
+        fn next(&mut self) -> Option<Self::Item> {
+            if self.relocation_index >= self.relocation_count {
+                return None;
+            }
+
+            if self.group_relocation_index >= self.group_relocation_count {
+                match self.read_group_fields() {
+                    Ok(_) => {}
+                    Err(e) => return Some(Err(e)),
+                }
+            }
+
+            match self.read_group_relocation() {
+                Ok(_) => Some(Ok(self.relocation.clone().into())),
+                Err(e) => Some(Err(e)),
+            }
+        }
+    }
+
+    impl<'data> Iterator for ParsingIterator<'data, false> {
+        type Item = Result<Rel, ParseError>;
```

Comment:
here, a different return value style from other iterators in the project.

when `&[8]` got unexpected eof, parsing `.rel` got `rela` and etc will get `Err(...)`